examples/tools.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+# typed: strong
+
+require_relative "../lib/anthropic"
+
+# gets API credentials from environment variable `ANTHROPIC_API_KEY`
+client = Anthropic::Client.new
+
+user_message = {
+  role: "user",
+  content: "What is the weather in SF?"
+}
+
+tools = [
+  {
+    name: "get_weather",
+    description: "Get the weather for a specific location",
+    input_schema: {
+      type: "object",
+      properties: {
+        location: {type: "string"}
+      }
+    }
+  }
+]
+
+message = client.messages.create(
+  model: "claude-sonnet-4-20250514",
+  max_tokens: 1024,
+  messages: [user_message],
+  tools: tools
+)
+
+puts "Initial response: #{JSON.pretty_generate(message)}"
+
+raise "Expected tool_use stop_reason" unless message.stop_reason == :tool_use
+
+tool = message.content.find { |c| c.type == :tool_use }
+
+raise "Tool use not found" unless tool
```

Comment:
This check isn't in the python version but it's necessary to prevent a type error on `tool.id` below.